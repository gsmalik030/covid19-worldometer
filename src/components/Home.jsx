import React from 'react'

const Home = () => {
  return (
    <header>
      <nav>
        
      </nav>
    </header>
  )
}

export default Home
