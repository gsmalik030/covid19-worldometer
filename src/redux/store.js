import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './country/countrySlice';
import detailsReducer from './details/detailsSlice';

const store = configureStore({
  reducer: {
    country: countryReducer,
    details: detailsReducer,
  },
});

export default store;
