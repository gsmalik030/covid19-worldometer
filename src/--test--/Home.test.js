import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../components/Home';

it('renders correctly', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});
