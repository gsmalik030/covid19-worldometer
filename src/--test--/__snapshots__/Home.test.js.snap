// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="world__info world__container"
  >
    <svg
      className="world-map"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M253.47 17.406C123.76 17.406 18.437 122.76 18.437 252.47c0 129.707 105.324 235.06 235.03 235.06 129.707 0 235.063-105.353 235.063-235.06 0-129.71-105.355-235.064-235.06-235.064zM367.874 68.75c61.246 38.19 101.97 106.14 101.97 183.72 0 17.143-1.993 33.823-5.75 49.81l-34.25-18.06 22 54.874c-9.454 21.647-22.362 41.432-38 58.687l-43.158-30.936-64.625 47.72-61.656 6.967-13.906-41.78-49.72 26.844-68.093-18.938 9.157 36.594c-28.41-21.793-51.23-50.466-66-83.563L81.25 304.47l32.25 17.124 59.22-9.875 2.843-40.908-37.344-1.718 4.905-17.844 30.78-25.313-25.093-15.625 67.22-38.593-45.345-29.657-66.625 40.187-49.437-15.28c13.812-32.14 35.21-60.22 61.906-82.064l-3.75 44.375 43.376-34.124 72 22.22-22.5-27.407L233 75.562l26.813 28.468 71 9.845-3.5-34.47 41.468 12.657-.905-23.312zm1.156 120.03L278 199.47l28.906 43.218 3.156 64.468L339.25 321l11.438-28.375 62.656 48.656L395.78 294l6.408-48.344-43.75-22.72 10.593-34.155zM221 192.438l-31.594 21.188 36.47 14.78 16.686-14.78L221 192.437zm22.188 144.688l18.687 52.594 19.78-42.564-38.467-10.03z"
      />
    </svg>
    <div>
      <h2
        className="country__name"
      >
        World
      </h2>
      <div
        className="country__facts"
      >
        <p
          className="country__population"
        >
          7.888 billion
        </p>
        <p
          className="country__continent"
        >
          682,048,548
        </p>
      </div>
    </div>
  </div>
  <div
    className="search__container"
  >
    <span>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
    </span>
    <input
      className="country__search"
      onChange={[Function]}
      placeholder="Eg: Afghanistan , South Africa"
    />
  </div>
  <div>
    <h1>
      STATS BY COUNTRY
    </h1>
  </div>
  <div>
    <ul
      className="container"
    />
  </div>
</section>
`;
